Fri Jan 20 07:40:20 CET 2012  Duncan Coutts <duncan@community.haskell.org>
  * Fix glib for change in RTS API in GHC 7.4
diff -rN -u old-glib/System/Glib/hsgclosure.c new-glib/System/Glib/hsgclosure.c
--- old-glib/System/Glib/hsgclosure.c	2012-02-13 23:00:36.783856552 +0100
+++ new-glib/System/Glib/hsgclosure.c	2012-02-13 23:00:36.879857028 +0100
@@ -107,8 +107,12 @@
     WHEN_DEBUG(g_debug("gtk2hs_closure_marshal(%p): about to rts_evalIO", hc->callback));

     /* perform the call */
+    #if __GLASGOW_HASKELL__>=704
+    rts_evalIO(&cap, rts_apply(CAP (HaskellObj)runIO_closure, call),&ret);
+    #else
     cap=rts_evalIO(CAP rts_apply(CAP (HaskellObj)runIO_closure, call),&ret);
-
+    #endif
+
     WHEN_DEBUG(g_debug("gtk2hs_closure_marshal(%p): about to rts_checkSchedStatus", hc->callback));

     /* barf if anything went wrong */
